### PR TITLE
switch to String.indexOf(int) in various BSD formats code

### PR DESCRIPTION
**Pull request by mtbc**
Inspired by https://github.com/openmicroscopy/bioformats/pull/2528#discussion_r76218340 -

> this is really just about switching to calling a method that can assume searching for a single character instead of having to care about search string length

this is a mechanical change so that a simpler library method can be used. It may not make much difference but sets a good example for others writing further code based on our existing code. https://ci.openmicroscopy.org/view/Bio-Formats/job/BIOFORMATS-DEV-merge-full-repository/ suffices for testing.

In the event of conflicts I'll remove the problem changes from this PR.

### CODE REVIEW

File: components/formats-bsd/src/loci/formats/ChannelSeparator.java [hunk 1]
```diff
@@ -119,7 +119,7 @@ public String getDimensionOrder() {
     String order = super.getDimensionOrder();
     if (reader.isRGB() && !reader.isIndexed()) {
       String newOrder = "XYC";
-      if (order.indexOf("Z") > order.indexOf("T")) newOrder += "TZ";
+      if (order.indexOf('Z') > order.indexOf("T")) newOrder += "TZ";
```
Comment (melissalinkert) [inline components/formats-bsd/src/loci/formats/ChannelSeparator.java]: Should this be ```order.indexOf('T')``` as well?

File: components/formats-bsd/src/loci/formats/FileStitcher.java [hunk 2]
```diff
@@ -1221,7 +1221,7 @@ protected void initReader(int sno, int fno) {
       String newOrder = ((DimensionSwapper) reader).getInputOrder();
       if ((externals[external].getFiles().length > 1 || !r.isOrderCertain()) &&
         (r.getRGBChannelCount() == 1 ||
-        newOrder.indexOf("C") == r.getDimensionOrder().indexOf("C")))
+        newOrder.indexOf('C') == r.getDimensionOrder().indexOf("C")))
```
Comment (melissalinkert) [inline components/formats-bsd/src/loci/formats/FileStitcher.java]: Should this be ```r.getDimensionOrder().indexOf('C')```?

**Comment by rleigh-codelibre** [comment]
This looks good to merge once the full-repo test for today is green (might be tomorrow before completion).

**Comment by sbesson** [comment]
Merging as per https://ci.openmicroscopy.org/view/Failing/job/BIOFORMATS-DEV-merge-full-repository/281/

**Pull request merged**